mapie/time_series_regression.py

```diff
@@ -0,0 +1,272 @@
+            ``(1-alpa+beta)-quantile - beta-quantile``.
+
+        Raises
+        ------
+        ValueError
+            If lower and upper bounds arrays don't have the same shape.
+        """
+        if lower_bounds.shape != upper_bounds.shape:
+            raise ValueError(
+                "Lower and upper bounds arrays should have the same shape."
+            )
+        alpha = cast(NDArray, alpha)
+        betas_0 = np.full(
+            shape=(len(lower_bounds), len(alpha)),
+            fill_value=np.nan,
+            dtype=float,
+        )
+
+        for ind_alpha, _alpha in enumerate(alpha):
+            betas = np.linspace(
+                _alpha / (len(lower_bounds) + 1),
+                _alpha,
+                num=len(lower_bounds),
+                endpoint=True,
+            )
+            one_alpha_beta = np_nanquantile(
+                upper_bounds,
+                1 - _alpha + betas,
+                axis=1,
+                method="higher",
+            )
+            beta = np_nanquantile(
+                lower_bounds,
+                betas,
+                axis=1,
+                method="lower",
+            )
+            betas_0[:, ind_alpha] = betas[
+                np.argmin(one_alpha_beta - beta, axis=0)
+            ]
+
+        return betas_0
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Compared to the method ``fit`` of ``MapieRegressor``, the ``fit``
+        method of ``MapieTimeSeriesRegressor`` computes the
+        ``conformity_scores_`` with relative values.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+        """
+        self = super().fit(X=X, y=y, sample_weight=sample_weight)
+        self.conformity_scores_ = self._relative_conformity_scores(X, y)
+        return self
+
+    def partial_fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Update the ``conformity_scores_`` attribute when new data with known
+        labels are available.
+        Note: Don't use ``partial_fit`` with samples of the training set.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples_test, n_features)
+            Input data.
+
+        y : ArrayLike of shape (n_samples_test,)
+            Input labels.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+
+        Raises
+        ------
+        ValueError
+            If the length of y is greater than the length of the training set.
+        """
+        X = cast(NDArray, X)
+        y = cast(NDArray, y)
+        n = len(self.conformity_scores_)
+        if len(X) > n:
+            raise ValueError(
+                "You try to update more residuals than there are!"
+            )
```

Comment:
Change it to a more polite sentence : 
"The number of observations to update is higher than the number of training instances."

Reply:
Done